src/rust/src/file_reader.rs

```diff
@@ -29,7 +29,10 @@ impl<'a, R: Read + Seek> FgbReader<'a, R> {
     pub fn open(reader: &'a mut R) -> Result<Self> {
         let mut magic_buf: [u8; 8] = [0; 8];
         reader.read_exact(&mut magic_buf)?;
-        if magic_buf != MAGIC_BYTES {
+        if magic_buf[0..2] != MAGIC_BYTES[0..2]
```

Comment:
```suggestion
        if magic_buf[0..=2] != MAGIC_BYTES[0..=2]
```

This part seems off (and similarly below).